docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+* Handling common misspellings and typos: Synonyms can be used to make sure that frequent or specific misspellings are correctly handled.
+
+Synonyms are grouped together using *synonyms sets*.
+You can have as many synonyms sets as you need.
+
+In order to use synonyms sets in {es}, you need to:
+
+* <<synonyms-define-synonyms>>
+* <<synonyms-store-synonyms>>
+* <<synonyms-synonym-analyzers>>
+
+[discrete]
+[[synonyms-define-synonyms]]
+=== Define synonyms sets
+
+Synonyms in a synonyms set are defined using *synonym rules*.
+Each synonym rule contains words that are synonyms.
+
+You can use two formats to define synonym rules: Solr and WordNet.
+
+[discrete]
+[[synonyms-define-synonyms-solr]]
+==== Solr format
+
+This format uses two different definitions:
+
+* Equivalent synonyms: Define groups of words that are equivalent. Words are separated by commas. Example:
+[source,synonyms]
+----
+ipod, i-pod, i pod
+computer, pc, laptop
+----
+* Explicit mappings: Matches a group of words to other words. Words on the left hand side of the rule definition are expanded into all the possibilities described on the right hand side. Example:
+
+[source,synonyms]
+----
+personal computer => pc
+sea biscuit, sea biscit => seabiscuit
+----
+
+[discrete]
+[[synonyms-define-synonyms-wordnet]]
+==== WordNet format
+
+https://wordnet.princeton.edu/[WordNet] defines synonyms sets spanning multiple lines. Each line contains the following information:
+
+* Synonyms set numeric identifier
+* Ordinal of the synonym in the synonyms set
+* Synonym word
+* Word type identifier: Noun (n), verb (v), adjective (a) or adverb (b).
+* Depth of the word in the synonym net
+
+The following example defines a synonym set for the words "come", "advance" and "approach":
+
+[source,synonyms]
+----
+s(100000002,1,'come',v,1,0).
+s(100000002,2,'advance',v,1,0).
+s(100000002,3,'approach',v,1,0).""";
+----
+
+[discrete]
+[[synonyms-store-synonyms]]
+=== Store your synonyms set
+
+Your synonyms sets need to be stored in {es} so your analyzers can refer to them.
+There are three ways to include your synonyms sets:
+
+[discrete]
+[[synonyms-store-synonyms-api]]
+==== Synonyms API
+
+You can use the <<synonyms-apis,synonyms APIs>> to manage synonyms sets.
+This is the most flexible approach, as it allows to dynamically define and modify synonyms sets.
+
+Changes in your synonyms sets will automatically reload the associated analyzers.
+
+[discrete]
+[[synonyms-store-synonyms-file]]
+==== Synonyms File
+
+You can store your synonyms set in a file.
+
+A synonyms set file needs to be uploaded to all your cluster nodes, and be located in the configuration directory for your {es} distribution.
+If you're using {ess}, you can upload synonyms files using {cloud}/ec-custom-bundles.html[custom bundles].
+
```

Comment:
I think it is better to remove this line about ESS. Soon in Serverless there will no option to use synonym files.

Reply:
ESS won't go away with serverless. I think it's good to keep this line as it can be cumbersome for users to find how to deal with synonyms in ESS.